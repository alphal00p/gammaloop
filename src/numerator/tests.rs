use symbolica::printer::PrintOptions;

use crate::tests_from_pytest::load_amplitude_output;

#[test]
#[allow(unused)]
fn lbl() {
    let (model, amplitude) = load_amplitude_output("./src/test_resources/lbl/");

    model
        .export_coupling_replacement_rules("./ignore/", PrintOptions::mathematica())
        .unwrap();
    let mut graph = amplitude.amplitude_graphs[0].graph.clone();

    graph.generate_numerator(&model);

    let mut numerator = graph.derived_data.numerator.clone().unwrap();
    // println!("{}", numerator);
    for (lhs, rhs) in graph.generate_lmb_replacement_rules() {
        numerator =
            lhs.into_pattern()
                .replace_all(numerator.as_view(), &rhs.into_pattern(), None, None);
    }

    println!("{}", numerator);

    // if let AtomView::Mul(mul) = numerator.as_view() {
    //     let net = SymbolicTensor::mul_to_tracking_network(mul).unwrap();

    //     println!("{}", net.dot());
    //     for (_, t) in net.graph.nodes {
    //         // println!("{}", t.structure());
    //     }
    // }
}
