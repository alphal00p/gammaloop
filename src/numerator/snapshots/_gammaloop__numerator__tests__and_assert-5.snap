---
source: src/numerator/tests.rs
expression: g_simp.to_canonical_string()
---
-16*Q(4,mink(4,27))*Q(5,mink(4,27))*Q(6,mink(4,29))*Q(7,mink(4,29))*p(mink(4,2))^2*phat^-2+-16*Q(4,mink(4,27))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,27))*p(mink(4,2))^2*phat^-2+-16*Q(4,mink(4,3))*Q(5,mink(4,28))*Q(6,mink(4,2))*Q(7,mink(4,28))*p(mink(4,2))*p(mink(4,3))*phat^-2+-24*Q(4,mink(4,27))*Q(5,mink(4,4))*Q(6,mink(4,4))*Q(7,mink(4,27))+-32*Q(4,mink(4,27))*Q(5,mink(4,27))*Q(6,mink(4,2))*Q(7,mink(4,2))+-32*Q(4,mink(4,27))*Q(5,mink(4,28))*Q(6,mink(4,27))*Q(7,mink(4,28))+-32*Q(4,mink(4,4))*Q(5,mink(4,4))*Q(6,mink(4,29))*Q(7,mink(4,29))+-8*Q(4,mink(4,2))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,2))+-8*Q(4,mink(4,2))*Q(5,mink(4,3))*Q(6,mink(4,29))*Q(7,mink(4,29))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,2))*Q(5,mink(4,4))*Q(6,mink(4,3))*Q(7,mink(4,4))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,2))*Q(5,mink(4,4))*Q(6,mink(4,4))*Q(7,mink(4,3))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,27))*Q(5,mink(4,2))*Q(6,mink(4,27))*Q(7,mink(4,2))+-8*Q(4,mink(4,27))*Q(5,mink(4,2))*Q(6,mink(4,27))*Q(7,mink(4,3))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,27))*Q(5,mink(4,2))*Q(6,mink(4,3))*Q(7,mink(4,27))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,27))*Q(5,mink(4,3))*Q(6,mink(4,27))*Q(7,mink(4,2))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,27))*Q(5,mink(4,4))*Q(6,mink(4,27))*Q(7,mink(4,4))*p(mink(4,2))^2*phat^-2+-8*Q(4,mink(4,3))*Q(5,mink(4,4))*Q(6,mink(4,4))*Q(7,mink(4,2))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,4))*Q(5,mink(4,28))*Q(6,mink(4,4))*Q(7,mink(4,28))+-8*Q(4,mink(4,4))*Q(5,mink(4,4))*Q(6,mink(4,2))*Q(7,mink(4,3))*p(mink(4,2))*p(mink(4,3))*phat^-2+-8*Q(4,mink(4,4))*Q(5,mink(4,4))*Q(6,mink(4,3))*Q(7,mink(4,2))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,2))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,3))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,2))*Q(5,mink(4,28))*Q(6,mink(4,3))*Q(7,mink(4,28))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,27))*Q(5,mink(4,27))*Q(6,mink(4,2))*Q(7,mink(4,3))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,27))*Q(5,mink(4,27))*Q(6,mink(4,3))*Q(7,mink(4,2))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,27))*Q(5,mink(4,28))*Q(6,mink(4,27))*Q(7,mink(4,28))*p(mink(4,2))^2*phat^-2+16*Q(4,mink(4,3))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,2))*p(mink(4,2))*p(mink(4,3))*phat^-2+16*Q(4,mink(4,4))*Q(5,mink(4,4))*Q(6,mink(4,2))*Q(7,mink(4,2))+24*Q(4,mink(4,27))*Q(5,mink(4,4))*Q(6,mink(4,27))*Q(7,mink(4,4))+32*Q(4,mink(4,27))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,27))+64*Q(4,mink(4,27))*Q(5,mink(4,27))*Q(6,mink(4,29))*Q(7,mink(4,29))+8*Q(4,mink(4,2))*Q(5,mink(4,4))*Q(6,mink(4,4))*Q(7,mink(4,2))+8*Q(4,mink(4,27))*Q(5,mink(4,3))*Q(6,mink(4,2))*Q(7,mink(4,27))*p(mink(4,2))*p(mink(4,3))*phat^-2+8*Q(4,mink(4,27))*Q(5,mink(4,4))*Q(6,mink(4,4))*Q(7,mink(4,27))*p(mink(4,2))^2*phat^-2+8*Q(4,mink(4,3))*Q(5,mink(4,2))*Q(6,mink(4,29))*Q(7,mink(4,29))*p(mink(4,2))*p(mink(4,3))*phat^-2+8*Q(4,mink(4,3))*Q(5,mink(4,4))*Q(6,mink(4,2))*Q(7,mink(4,4))*p(mink(4,2))*p(mink(4,3))*phat^-2+8*Q(4,mink(4,4))*Q(5,mink(4,2))*Q(6,mink(4,4))*Q(7,mink(4,2))+8*Q(4,mink(4,4))*Q(5,mink(4,28))*Q(6,mink(4,28))*Q(7,mink(4,4))+8*Q(4,mink(4,4))*Q(5,mink(4,4))*Q(6,mink(4,29))*Q(7,mink(4,29))*p(mink(4,2))^2*phat^-2
